bson/_cbsonmodule.c

```diff
@@ -2797,6 +2807,86 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    int32_t initial_position;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
+
+
+    if (!_get_buffer(arr, &view)) {
+        goto fail;
```

Comment:
There's a bug here. When this line fails we'll mistakenly call PyBuffer_Release(&view).

Reply:
Fixed

Reply:
The same bug now exists when pymongo_buffer_new fails.

Reply:
Fixed